apiserver/v1/types_deprecatedapirequest.go

```diff
@@ -0,0 +1,110 @@
+// Package v1 is an api version in the apiserver.openshift.io group
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+// +kubebuilder:subresource:status
+// +genclient:nonNamespaced
+
+// DeprecatedAPIRequest tracts requests made to a deprecated API. The instance name should
+// be of the form `resource.version.group`, matching the deprecated resource.
+type DeprecatedAPIRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines the characteristics of the resource.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec DeprecatedAPIRequestSpec `json:"spec"`
+
+	// Status contains the observed state of the resource.
+	Status DeprecatedAPIRequestStatus `json:"status,omitempty"`
+}
+
+type DeprecatedAPIRequestSpec struct {
+	// RemovedRelease is when the API will be removed.
+	// +kubebuilder:validation:MaxLength=64
+	// +required
+	RemovedRelease string `json:"removedRelease"`
+}
+
+// +k8s:deepcopy-gen=true
+type DeprecatedAPIRequestStatus struct {
+
+	// Conditions contains details of the current status of this API Resource.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// RequestsLastHour contains request history for the current hour. This is porcelain to make the API
+	// easier to read by humans seeing if they addressed a problem.
+	RequestsLastHour RequestLog `json:"requestsLastHour"`
+
+	// RequestsLast24h contains request history for the last 24 hours, indexed by the hour, so
+	// 12:00AM-12:59 is in index 0, 6am-6:59am is index 6, etc..
+	RequestsLast24h []RequestLog `json:"requestsLast24h"`
```

Comment:
is the current index up-to-date? Or is `requestLastHour` only copied over at the end of the hour?

Reply:
Current index will also be up-to-date.

Reply:
Maybe add "The index of the current hour is also updated live."

Reply:
Updated godoc.